external-import/socradar/src/lib/radar.py

```diff
@@ -0,0 +1,354 @@
+                description=f"Feed Provider: {maintainer_name}",
+                sectors=["technology"],
+                created=current_time,
+                modified=current_time,
+            )
+
+            # Store in mapping for reuse
+            self.identity_mapping[maintainer_name] = identity
+            self.helper.log_info(
+                f"Created new identity for maintainer: {maintainer_name}"
+            )
+            return identity
+
+        except Exception as e:
+            self.helper.log_error(
+                f"Error creating identity for {maintainer_name}: {str(e)}"
+            )
+            return None
+
+    def _process_feed_item(self, item, work_id):
+        try:
+            value = item["feed"]
+            feed_type = item.get("feed_type", "").lower()
+            maintainer = item.get("maintainer_name", "Unknown")
+            first_seen = datetime.strptime(item["first_seen_date"], "%Y-%m-%d %H:%M:%S")
+            last_seen = datetime.strptime(item["latest_seen_date"], "%Y-%m-%d %H:%M:%S")
+
+            if last_seen <= first_seen:
+                last_seen = first_seen + timedelta(days=365)
+
+            # Get or create identity for this maintainer
+            maintainer_identity = self._get_or_create_identity(maintainer)
+            if not maintainer_identity:
+                self.helper.log_error(
+                    f"Could not create identity for maintainer: {maintainer}"
+                )
+                return
+
+            pattern = self._create_stix_pattern(value, feed_type)
+            if not pattern:
+                self.helper.log_error(
+                    f"Could not create pattern for: {value} ({feed_type})"
+                )
+                return
+
+            # Create kill chain phase
+            kill_chain_phase = KillChainPhase(
+                kill_chain_name="lockheed-martin-cyber-kill-chain",
+                phase_name="reconnaissance",
+            )
+
+            indicator = Indicator(
+                id=f"indicator--{str(uuid.uuid4())}",
+                name=f"{feed_type.upper()}: {value}",
+                description=f"Type: {feed_type}\nValue: {value}\nSource: {maintainer}",
+                pattern_type="stix",
+                pattern=pattern,
+                valid_from=first_seen,
+                valid_until=last_seen,
+                labels=[feed_type, "malicious-activity"],
+                confidence=75,
+                indicator_types=self._get_indicator_type(feed_type),
+                kill_chain_phases=[kill_chain_phase],
+                created=first_seen,
+                modified=first_seen,
+                created_by_ref=maintainer_identity.id,  # Use maintainer's identity
+                object_marking_refs=[TLP_WHITE],
+            )
+
+            # Create relationship between indicator and maintainer identity
+            relationship = Relationship(
+                id=f"relationship--{str(uuid.uuid4())}",
```

Comment:
This would create stochastic id and will prevent de-duplication performed by the platform.

You should rather use pycti.StixCoreRelationship.generate_id method.